### PR TITLE
don't show force_hide selectors on cosmetic filters iterations

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Uplift of https://github.com/brave/brave-core/pull/8046
Resolves https://github.com/brave/brave-browser/issues/14294

## Submitter Checklist:

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by SergeyZhukovsky** [comment]
post-init failed due to known audit failure, not related to the current PR

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.21.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/8046#issuecomment-785424690.

**Pull request merged**
